### PR TITLE
fix: Add a panel to styleguide and use it

### PR DESCRIPTION
**Pull request by MaxPostema**
Adds a Panel to the styleguide. ( replaces some random and inconsistent background white divs )
This makes sure that every white block has the same padding. 

Changes the look and feel of the forms, tableExplorer and schema editor

### CODE REVIEW

File: apps/styleguide/src/layout/Panel.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    :class="{'m-n2': outside}"
+  >
+    <slot />
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    outside: {
```
Comment (chinook25) [inline apps/styleguide/src/layout/Panel.vue:13]: would outsidePadding not be a better name for this?

File: apps/styleguide/src/tables/TableExplorer.vue [hunk 2]
```diff
@@ -179,29 +189,29 @@
...
+import ButtonAlt from '../forms/ButtonAlt';
+import SelectionBox from './SelectionBox';
+import ButtonDropdown from '../forms/ButtonDropdown';
+import InputSelect from '../forms/InputSelect';
+import TableCards from './TableCards';
+import IconAction from '../forms/IconAction';
+import RecordCard from './RecordCard';
+import TableSettings from './TableSettings';
+import Panel from '../layout/Panel';
+const View = {TABLE: 'table', CARDS: 'cards', RECORD: 'record', EDIT: 'edit'};
```
Comment (chinook25) [inline apps/styleguide/src/tables/TableExplorer.vue:214]: keeping an enter after the imports would be nice :)

**Review by chinook25 — APPROVED** [review]

File: apps/styleguide/src/layout/Panel.vue (hunk 1 above)
Comment (MaxPostema) [reply apps/styleguide/src/layout/Panel.vue:13]: agreed

File: apps/styleguide/src/tables/TableExplorer.vue (hunk 2 above)
Comment (MaxPostema) [reply apps/styleguide/src/tables/TableExplorer.vue:214]: Yes, changed

**Review by mswertz — CHANGES_REQUESTED** [review]
In the old behavior of 'schema' the left panel was sticky so I table of contents scrolls independent of right panel.

**Comment by mswertz** [comment]
I don't feel this addition of panel is an improvement.

**Comment by MaxPostema** [comment]
> In the old behavior of 'schema' the left panel was sticky so I table of contents scrolls independent of right panel. See https://emx2.dev.molgenis.org/pet%20store/schema/#/

Breaking the left panel was an unintended side effect. It has been fixed.

> In addition I am not happy with our codestyle. Please change ' back to " because now massive unneeded changes in track change history.

prettierrc.js has been changed to reflect this. ( I believe that the single quote was chosen to conform to the molgenis developer guide. )

**Comment by MaxPostema** [comment]
> I don't feel this addition of panel is an improvement.

There are white boxes throughout the application to contain visual elements that group together. Currently they have different paddings in some cases like the 'old' images below no padding at all. Some have rounded edges, others have sharp edges. By adding a panel I want to give developers a way to group components together in a more consistent way ensuring the same padding/style.

Old: (there is a padding to the left of the "table", "category" heading, but not on the top of the headings)
Schermafbeelding 2022-03-21 om 09 20 22

New: (padding in this panel is the same)
Schermafbeelding 2022-03-21 om 09 20 31

Old: (no padding at all here, the username and password text are touching the white border)
Schermafbeelding 2022-03-21 om 09 27 01

New: ( panel with same padding as in other places groups. Also grouped the form together as one element  )
Schermafbeelding 2022-03-21 om 09 26 39

**Comment by mswertz** [comment]
> > I don't feel this addition of panel is an improvement.
> 
> There are white boxes throughout the application to contain visual elements that group together. Currently they have different paddings in some cases like the 'old' images below no padding at all. Some have rounded edges, others have sharp edges. By adding a panel I want to give developers a way to group components together in a more consistent way ensuring the same padding/style.
> 
> Old: (there is a padding to the left of the "table", "category" heading, but not on the top of the headings) Schermafbeelding 2022-03-21 om 09 20 22
> 
> New: (padding in this panel is the same) Schermafbeelding 2022-03-21 om 09 20 31
> 
> Old: (no padding at all here, the username and password text are touching the white border) Schermafbeelding 2022-03-21 om 09 27 01
> 
> New: ( panel with same padding as in other places groups. Also grouped the form together as one element ) Schermafbeelding 2022-03-21 om 09 26 39

Okay, i see added value of what you show here. 
So as long as we fix the other issues with the sticky panel and the spacing with the inputs we can be happy?

**Review by mswertz — APPROVED** [review]

**Comment by MaxPostema** [comment]
It has some issues with destroying other parts of the code. Closing for now.

**Pull request closed without merging**